content/courses/Ada_For_The_C_Embedded_Developer/chapters/06_Translation.rst

```diff
@@ -1395,16 +1395,145 @@ a byte-aligned pointer. Then, it simply copies the data byte-by-byte.
 .. _OverlaysVsUncheckedConversions:
 
 Overlays vs. Unchecked Conversions
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. todo::
+Unchecked conversions are another way of converting between unrelated data
+types. This conversion is done by instantiating the generic
+:ada:`Unchecked_Conversions` function for the types you want to convert. Let's
+look at a simple example:
 
-    Complete section!
+[Ada]
+
+.. code:: ada run_button project=Courses.Ada_For_C_Embedded_Dev.Translation.Simple_Unchecked_Conversion
+
+    with Ada.Text_IO;              use Ada.Text_IO;
+    with Ada.Unchecked_Conversion;
+
+    procedure Simple_Unchecked_Conversion is
+       type State is (Off, State_1, State_2)
+         with Size => Integer'Size;
+
+       for State use (Off => 0, State_1 => 32, State_2 => 64);
+
+       function To_Integer is new Ada.Unchecked_Conversion (Source => State,
```

Comment:
Just an aside. Consider using a naming scheme of "As_*" instead of "To_*" because we aren't really converting the value (unless we then assign the result of the conversion to something else). Instead, we are asking the compiler to treat the value as if it is of a different type, so that we can manipulate/query it in a manner consistent with that other type. For example:

type Int_Pointer is access all Integer;
function As_Int_Pointer is new Ada.Unchecked_Conversion (Source => System.Address, Target => Int_Pointer);

Then, given an address X, hopefully of some integer, we can say As_Int_Pointer (X).all ;= 42; -- the answer to life, the universe, ...